schema/validator.go

```diff
@@ -44,6 +52,13 @@ func (v Validator) Validate(src io.Reader) error {
 		return errors.Wrap(err, "unable to read the document file")
 	}
 
+	if f, ok := mapValidateDescendants[v]; ok && f != nil {
```

Comment:
There's no reason to check both `ok` and `nil` if you're silently ignoring the `nil` case.  I think you should either [skip the `nil` check and pacnic if someone stuck a `nil` value in `mapValidateDescendants`][1] or [explicitly check for and error out if someone stuck a `nil` value in `mapValidateDescendants`][2]

[1]: https://github.com/opencontainers/image-spec/pull/341#discussion_r82013412
[2]: https://github.com/opencontainers/image-spec/pull/341#discussion_r82113598